MosipNexus/api/main.py

```
@@ -0,0 +1,529 @@
+
+
+@app.post("/chat", response_model=ChatResponse, tags=["Chat"],
+          summary="Ask a MOSIP question",
+          description="Main RAG endpoint. Returns answer + sources + confidence. Pass `session_id` from the response back to maintain conversation context.")
+def chat(req: ChatRequest) -> ChatResponse:
+    session_id = req.session_id or str(uuid.uuid4())
+    memory = _get_session(session_id)
+
+    try:
+        result = ask(req.question, memory.messages, req.language)
+    except Exception as e:
```

Ruff (0.15.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)